lib/api/bucketPutWebsite.js

```diff
@@ -0,0 +1,60 @@
+import { errors } from 'arsenal';
+import async from 'async';
+
+import services from '../services';
+import { parseWebsiteConfigXml } from './apiUtils/bucket/bucketWebsite';
+import metadata from '../metadata/wrapper';
+
+/**
+ * Bucket Put Website - Create bucket website configuration
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+export default function bucketPutWebsite(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketPutWebsite' });
+    const bucketName = request.bucketName;
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketPutWebsite',
+        log,
+    };
+    return async.waterfall([
+        function parseXmlBody(next) {
+            if (request.post) {
+                log.trace('parsing website configuration from request body');
+                return parseWebsiteConfigXml(request.post, log,
+                    (err, config) => next(err, config));
+            }
+            // If no request body at all
+            return next(errors.MissingRequestBodyError);
+        },
+        function validateBucketAuthorization(config, next) {
+            services.metadataValidateAuthorization(metadataValParams,
```

Comment:
Instead of using `services.metadataValidateAuthorization` I would recommend calling the logic needed inside it directly, as we are planning to phase out that method with refactors for other APIs.
You can look at https://github.com/scality/S3/pull/434/files for reference.